stable/kong/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) .Values.ingressController.enabled .Values.ingressController.metrics.serviceMonitor.create }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ template "kong.fullname" . }}
+  {{- if .Values.ingressController.metrics.serviceMonitor.namespace }}
+  namespace: {{ .Values.ingressController.metrics.serviceMonitor.namespace }}
+  {{- end }}
+  labels:
+    app: {{ template "kong.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
+  endpoints:
+  - port: kong-admin
```

Comment:
With #15079 merged in, port 9542 will be used to export metrics. 
let's use that.
Also, it works only on HTTP scheme.

Reply:
@hbagdi Does a custom-server-block, introduced in #15079 enables the Prometheus plugin? It seems like so.

Reply:
Sort of.
It won't enable the plugin inside Kong. But it will expose the default metrics that are available even when the plugin is not enabled for any specific service.

Reply:
Right now, with the `custom-server-block` present by default, this PR will only optionally add a `ServiceMonitor`